wrappers/desktop_test.go

```diff
@@ -128,34 +169,114 @@ func (s *iconsTestSuite) TestEnsurePackageDesktopFilesNilSnapInfo(c *C) {
 	c.Assert(wrappers.EnsureSnapDesktopFiles([]*snap.Info{nil}), ErrorMatches, "internal error: snap info cannot be nil")
 }
 
-func (s *desktopSuite) TestRemovePackageDesktopFiles(c *C) {
-	mockDesktopFilePath := filepath.Join(dirs.SnapDesktopFilesDir, "foo_foobar.desktop")
+func (s *desktopSuite) TestEnsurePackageDesktopFilesExistingFileError(c *C) {
+	const desktopAppYaml = `
+name: foo
+version: 1.0
+plugs:
+  desktop:
+    desktop-file-ids: [org.example.Foo]
+`
+	info := snaptest.MockSnap(c, desktopAppYaml, &snap.SideInfo{Revision: snap.R(11)})
+	c.Assert(info.Plugs["desktop"], NotNil)
+
+	// Mock existing desktop file with same name for another snap
+	var mockBadDesktopFile = []byte(`
+[Desktop Entry]
+Name=foo
+X-SnapInstanceName=bar`)
+	badDesktopFilePath := filepath.Join(dirs.SnapDesktopFilesDir, "org.example.Foo.desktop")
+	c.Assert(os.MkdirAll(dirs.SnapDesktopFilesDir, 0755), IsNil)
+	c.Assert(os.WriteFile(badDesktopFilePath, mockBadDesktopFile, 0644), IsNil)
+
+	// generate .desktop file in the package baseDir
+	baseDir := info.MountDir()
+	c.Assert(os.MkdirAll(filepath.Join(baseDir, "meta", "gui"), 0755), IsNil)
+	c.Assert(os.WriteFile(filepath.Join(baseDir, "meta", "gui", "org.example.Foo.desktop"), mockDesktopFile, 0644), IsNil)
+
+	err := wrappers.EnsureSnapDesktopFiles([]*snap.Info{info})
+	c.Assert(err, ErrorMatches, `cannot install "org.example.Foo.desktop": ".*/var/lib/snapd/desktop/applications/org.example.Foo.desktop" already exists for another snap`)
+
+	c.Check(badDesktopFilePath, testutil.FileEquals, mockBadDesktopFile)
+}
+
+func (s *desktopSuite) testRemovePackageDesktopFiles(c *C, triggerErr bool) {
+	var desktopFileTemplate = `
```

Comment:
It is better to use `const` for format strings. If you do that, then `go vet` will give you warnings when you make mistakes, rather than the string ending up weird at runtime.

```
phelps@arch ~/tmp> cat main.go
package main

import "fmt"

func main() {
        const template = "hello, %d\n"

        name := "andrew"
        fmt.Printf(template, name)
}
phelps@arch ~/tmp> go vet
# m
# [m]
./main.go:9:2: fmt.Printf format %d has arg name of wrong type string
```

Reply:
cool, I didn't know that